packages/web-api-send-notification-runner/src/sender/notification-sender.spec.ts

```diff
@@ -14,30 +27,169 @@ class MockableLogger extends Logger {}
+    let onDemandPageScanRunResultProviderMock: IMock<OnDemandPageScanRunResultProvider>;
+    let webAPIMock: IMock<NotificationSenderWebAPIClient>;
     let notificationSenderMetadataMock: IMock<NotificationSenderConfig>;
     let loggerMock: IMock<MockableLogger>;
+    let serviceConfigMock: IMock<ServiceConfiguration>;
+    let systemMock: IMock<typeof System>;
+    let processStub: typeof process;
+    let scanConfig: ScanRunTimeConfig;
+
     const notificationSenderMetadata: NotificationSenderMetadata = {
-        id: 'id',
+        scanId: 'id',
         scanNotifyUrl: 'scanNotifyUrl',
+        runStatus: 'completed',
+        scanStatus: 'pass',
+    };
+
+    const onDemandPageScanResult: OnDemandPageScanResult = {
+        url: 'url',
+        scanResult: null,
+        reports: null,
+        run: {
+            state: 'queued' as OnDemandPageScanRunState,
+            timestamp: 'timestamp',
+        },
+        priority: 1,
+        itemType: ItemType.onDemandPageScanRunResult,
+        id: 'id',
+        partitionKey: 'item-partitionKey',
+        batchRequestId: 'batch-id',
     };
 
     beforeEach(() => {
+        scanConfig = {
+            failedPageRescanIntervalInHours: 3,
+            maxScanRetryCount: 4,
+            maxSendNotificationRetryCount: 3,
+            minLastReferenceSeenInDays: 5,
+            pageRescanIntervalInDays: 6,
+            accessibilityRuleExclusionList: [],
+            scanTimeoutInMin: 1,
+        };
+        onDemandPageScanRunResultProviderMock = Mock.ofType(OnDemandPageScanRunResultProvider, MockBehavior.Strict);
+        webAPIMock = Mock.ofType(NotificationSenderWebAPIClient);
         loggerMock = Mock.ofType(MockableLogger);
+        serviceConfigMock = Mock.ofType(ServiceConfiguration);
+        serviceConfigMock
+            .setup(async s => s.getConfigValue('scanConfig'))
+            .returns(async () => scanConfig)
+            .verifiable(Times.once());
+        systemMock = Mock.ofInstance(System, MockBehavior.Strict);
         notificationSenderMetadataMock = Mock.ofType(NotificationSenderConfig);
         notificationSenderMetadataMock.setup(s => s.getConfig()).returns(() => notificationSenderMetadata);
 
-        sender = new NotificationSender(notificationSenderMetadataMock.object, loggerMock.object);
+        processStub = {} as typeof process;
+        processStub.env = { batchJobId: 'job 1' };
+
+        sender = new NotificationSender(
+            onDemandPageScanRunResultProviderMock.object,
+            webAPIMock.object,
+            notificationSenderMetadataMock.object,
+            loggerMock.object,
+            serviceConfigMock.object,
+            processStub,
+            systemMock.object,
+        );
+    });
+
+    it('Send Notification Succeeded', async () => {
+        setupReadScanResultCall(onDemandPageScanResult);
+        onDemandPageScanResult.notification = generateNotification(notificationSenderMetadata.scanNotifyUrl, 'sent', null);
```

Comment:
this doesnt seem to be testing properly. create a clone object & then set notification in that object & validate that update call is made against object equivalent to the cloned object. same for other tests